### PR TITLE
Initial development

### PR DESCRIPTION
**Pull request by p-tombez**

### CODE REVIEW

File: wrappers/tempo_client.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        params = {
+            'jira_url': self.jira_url,
+            'tempo_api': self.tempo_api,
+            'date_from': date_window.start.date().isoformat(),
+            'date_to': date_window.stop.date().isoformat(),
+            'tempo_user': self.tempo_user,
+        }
+
+        url = "{jira_url}{tempo_api}worklogs" \
+              "?username={tempo_user}&dateFrom={date_from}&dateTo={date_to}"
```
Comment (guewen) [inline wrappers/tempo_client.py:50]: ~~So you didn't solve the pagination problem you blame on the other script? :confused:~~

Unlikely to hit 50 worklogs within the window for one user

File: exporter.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            to_create.append(l)
+
+    Utils.report(to_create, to_delete, attendances)
+
+    if args.no_interactive or Utils.ask_confirmation():
+        for l in to_create:
+            tempo.create_worklog(l)
+
+        for l in to_delete:
+            tempo.delete_worklog(l)
```
Comment (guewen) [inline exporter.py:198]: Why not update the entries when only the duration changes?

File: wrappers/tempo_client.py (hunk 1 above)
Comment (p-tombez) [reply wrappers/tempo_client.py:50]: There is no pagination in the Tempo API

File: exporter.py (hunk 2 above)
Comment (p-tombez) [reply exporter.py:198]: That's on the roadmap ;)
I didn't have enough time to enhance it further

**Comment by guewen** [comment]
The default week when no `-w` is provided is the next week instead of the current one, so nothing is uploaded.

**Comment by p-tombez** [comment]
That's the issue with Python week number representation.
For Python, everything that is before the first Monday of the year is week 0, then it increments. E.g. this year, week 1 for python is 7-13 January. So, if you look at your calendar, use the week number corresponding to the week you want to sync, it will not work.
That's why I recommend using `-w $(date +"%W")` to have the correct number.
Some correction could be done (but, we cannot just put `- 1` to the value, because it will not work when 1st of January is a Monday), but we would have to adapt the call if we use the helper I mention earlier

**Comment by guewen** [comment]
Can we use `strftime` instead?

```
>>> datetime.datetime.now().strftime("%W"), datetime.date.today().isocalendar()[1]
('15', 16)
```

**Comment by guewen** [comment]
I have to say, I have had a way better experience using https://github.com/ProgrammersOfVilnius/gtimelog2jira. Why, besides it just worked fine from the start. Both with vilnius' and this one, I have had entries not pushed to Jira because the task is not found (BTW I thought this script would report them as I thought it was a reason not to use the other one). However, it was rather easy to find the missing entries with vilnius' and a PITA with this script (I didn't figure out yet what entries I miss in my >20 entries for yesterday), because:

* all entries are stacked at 00:00 (vilnius respect the same start time, and then it's easy to follow through the entries)
* having `project: TASK-X description` in gtimelog entries which not only make them painful to write (it's useless IMO `TASK-X: description` is more reasonable), make them hard to read when trying to visually match what we have in tempo and in gtimelog 

Can you give me reasons for spending time on this script?

**Comment by guewen** [comment]
>  (I didn't figure out yet what entries I miss in my >20 entries for yesterday)

Impossible to find the missing entry with this script. I dropped all my worklogs, re-imported everything using ProgrammersOfVilnius/gtimelog2jira and could find in 2 minutes following the order of the worklogs...

**Comment by p-tombez** [comment]
First of all, I have to say that I was forced to push the script in its current state, meaning it's not optimized, nor user friendly in matter of error reporting and is missing some key feature (all reported in the roadmap chapter of the readme).
That being said, let's focus on your comments
* "All entries are stacked at 00:00", yes so as described in the card, worklogs should be aggregated, so start/end time cannot be respected
* "Gtimlog format", you can use both notations in GTimeLog. It was asked to be able to group the entries by project in gtimelog, hence the notation "project-x: TASK-X description", but you are free to use "TASK-X description" if you like

**Comment by guewen** [comment]
>     * "All entries are stacked at 00:00", yes so as described in the card, worklogs should be aggregated, so start/end time cannot be respected

It was also written that we should use the existing script rather than writing a new one :stuck_out_tongue_winking_eye:. Honestly I would question this requirement. Might seem a good idea at first, then it's a pain when we want to find differences.

>     * It was asked to be able to group the entries by project in gtimelog

Who asked that? I mean, we can group in jira, in odoo...



>     * "project-x: TASK-X description", but you are free to use "TASK-X description" if you like

Ok thanks, didn't saw this.

**Comment by simahawk** [comment]
we must be able to use `TASK-X: description`, which BTW it should be the only allowed way of doing it since we always want to hit a specific task.
This does NOT work now. All my $PROJ-$X entries are completely ignored :sob:

**Comment by simahawk** [comment]
I merge and open PRs to fix

**Pull request merged**